compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -549,7 +549,11 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
 
             match source_file.name {
                 FileName::Real(ref original_file_name) => {
-                    let adapted_file_name = if self.tcx.sess.should_prefer_remapped_for_codegen() {
+                    let adapted_file_name = if self
+                        .tcx
+                        .sess
+                        .should_prefer_remapped(RemapPathScopeComponents::DEBUGINFO)
```

Comment:
This is kind of an unclear case to me for a number of reasons:

- Crate metadata is not really "debuginfo". One could expect it to be covered by the "object" scope.
- We don't know for certain how the source file paths stored here will be used by downstream compilation sessions. There's nothing preventing the compiler from emitting them to something in the macro or diagnostics scope. 

I think we should err on the side of caution and remap unconditionally (maybe revisit the question later?)

Reply:
Sure. Done.